### PR TITLE
bugfix/ 20729-a11y-mirror-mouseclick-for-keyboard

### LINKED ISSUE #20729: Highcharts accessibility for event click() is missing
#### Expected behaviour
Click event should have the same `event` context.

#### Actual behaviour
If you click on the point with a mouse you get a different result than with pressing enter on the keyboard.

#### Live demo with steps to reproduce
[demo](https://jsfiddle.net/BlackLabel/x091h5pb/show)
[SO thread](https://stackoverflow.com/questions/78071398/highcharts-accessibility-enter-and-click-events-function-mouseover-is-not-enoug)

#### Product version
Highcharts JS v11.3.0 (2024-01-10)

### PR DESCRIPTION
**Pull request by MarkusBarstad**
Fixed #20729, pressing "enter" on a tabbed point resulted in different event than clicking it with a mouse.

### CODE REVIEW

**Review by hubertkozik — CHANGES_REQUESTED** [review]
The fix looks good to me, both events (pointer and keyboard) have the same properties now (`chartX`, `chartY`, `isTrusted`, `point`). I wouldn't add more, but the thing I would add is an unit-test. We're already testing the keyboard navigation in `unit-tests/accessibility/accessibility-options` so maybe it's a good place to add it?

____
Also, the Visual comparison is broken, maybe merge master would help?

**Comment by MarkusBarstad** [comment]
> The fix looks good to me, both events (pointer and keyboard) have the same properties now (`chartX`, `chartY`, `isTrusted`, `point`). I wouldn't add more, but the thing I would add is an unit-test. We're already testing the keyboard navigation in `unit-tests/accessibility/accessibility-options` so maybe it's a good place to add it?
> 
> Also, the Visual comparison is broken, maybe merge master would help?

@hubertkozik Thanks for feedback, will do!

File: ts/Accessibility/Components/SeriesComponent/SeriesKeyboardNavigation.ts [hunk 1]
```diff
@@ -421,9 +421,19 @@ class SeriesKeyboardNavigation {
...
+                            const { plotLeft, plotTop } = this.chart,
+                                { plotX = 0, plotY = 0 } = point;
+
+                            event = {
+                                ...(event as any),
+                                chartX: plotLeft + plotX,
+                                chartY: plotTop + plotY,
+                                point: point,
+                                target: point.graphic?.element
+                            };
```
Comment (hubertkozik) [inline ts/Accessibility/Components/SeriesComponent/SeriesKeyboardNavigation.ts:433]: What do you think about creating a new file `KeyboardEvent.d.ts` next to `PointerEvent.d.ts`, which will export the extended interface of `KeyboardEvent` (just like it is now for `PointerEvent`), then we can import this type into `SeriesKeyboardNavigation.ts` and the types will be correct, without any 'any's. That would also mean, that the type of event argument of `firePointEvent` would need to change (extended `KeyboardEvent` type).

TL;DR; Create a new extended interface `KeyboardEvent` same as the existing `PointerEvent`.

**Review by hubertkozik — CHANGES_REQUESTED** [review]
The test looks good! After latest commit I've got an idea how to do it even better - comment below👇

File: ts/Accessibility/Components/SeriesComponent/SeriesKeyboardNavigation.ts (hunk 1 above)
Comment (bre1470) [reply ts/Accessibility/Components/SeriesComponent/SeriesKeyboardNavigation.ts:433]: This would make things consistent. Good idea! 👍

Comment (MarkusBarstad) [reply ts/Accessibility/Components/SeriesComponent/SeriesKeyboardNavigation.ts:433]: Sounds good!

File: ts/Core/KeyboardEvent.d.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * */
+
+import type Point from './Series/Point';
+
+export interface KeyboardEvent extends globalThis.KeyboardEvent {
+    chartX: number;
+    chartY: number;
+    point?: Point;
+}
```
Comment (bre1470) [inline ts/Core/KeyboardEvent.d.ts:23]: Default export is missing.

File: ts/Accessibility/Components/SeriesComponent/SeriesKeyboardNavigation.ts [hunk 3]
```diff
@@ -22,6 +22,7 @@ import type Accessibility from '../../Accessibility';
 import type {
     AccessibilityKeyboardNavigationSeriesNavigationOptions
 } from '../../Options/A11yOptions';
+import type { KeyboardEvent } from '../../../Core/KeyboardEvent';
```
Comment (bre1470) [inline ts/Accessibility/Components/SeriesComponent/SeriesKeyboardNavigation.ts:25]: Import default.

**Review by bre1470 — CHANGES_REQUESTED** [review]
Technically this looks good, and some keyboard issues are solved for me. 👍

One consistency thing is left to do.

**Review by bre1470 — APPROVED** [review]

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!
